test/GogeDao.t.sol

```diff
@@ -46,49 +45,87 @@ contract DaoTest is Utility, Test {
+        metadata.description = "This is a mock poll, for testing";
+        metadata.endTime = block.timestamp + 5 days;
+
+        // create poll
+        gogeToken.approve(address(gogeDao), gogeDao.minAuthorBal());
+        gogeDao.createPoll(GogeDAO.PollType.other, metadata);
+
+        // Verify state change
+        assertEq(gogeDao.pollNum(), _pollNum + 1);
+        assertEq(gogeDao.getMetadata(gogeDao.pollNum()).description, "This is a mock poll, for testing");
+    }
+
+
+    // ~~ Unit Tests ~~
+
+    /// @notice Verify the ability for holders of gogeToken to create polls on gogeDao
     function test_gogeDao_createPoll() public {
         // create poll metadata
         GogeDAO.Metadata memory metadata;
         metadata.description = "I want to add Joe to the naughty list";
-        metadata.time2 = block.timestamp + 2 days;
+        metadata.endTime = block.timestamp + 2 days;
         metadata.addr1 = address(joe);
         metadata.boolVar = true;
 
         // create poll
+        gogeToken.approve(address(gogeDao), gogeDao.minAuthorBal());
         gogeDao.createPoll(GogeDAO.PollType.modifyBlacklist, metadata);
 
         // Verify state change
         assertEq(gogeDao.pollNum(), 1);
+        assertEq(gogeDao.pollAuthor(1), address(this));
         assert(gogeDao.pollTypes(1) == GogeDAO.PollType.modifyBlacklist);
 
-        assertEq(
-            gogeDao.getMetadata(1).description,
-            "I want to add Joe to the naughty list"
-        );
-        assertEq(gogeDao.getMetadata(1).time1, block.timestamp);
-        assertEq(gogeDao.getMetadata(1).time2, block.timestamp + 2 days);
+        assertEq(gogeDao.getMetadata(1).description, "I want to add Joe to the naughty list");
+        assertEq(gogeDao.getMetadata(1).endTime, block.timestamp + 2 days);
         assertEq(gogeDao.getMetadata(1).addr1, address(joe));
         assertEq(gogeDao.getMetadata(1).boolVar, true);
 
-        // Emit poll data
-        // emit log_string  (gogeDao.getMetadata(1).description);
-        // emit log_uint    (gogeDao.getMetadata(1).time1);
-        // emit log_uint    (gogeDao.getMetadata(1).time2);
-        // emit log_address (gogeDao.getMetadata(1).addr1);
-        // emit log_bool    (gogeDao.getMetadata(1).boolVar);
+        assertEq(gogeDao.polls(1, address(this)), gogeDao.minAuthorBal());
+        assertEq(gogeDao.totalVotes(1), gogeDao.minAuthorBal());
+        assertEq(gogeToken.balanceOf(address(gogeDao)), gogeDao.minAuthorBal());
+
+        address[] memory voters = gogeDao.getVoterLibrary(1);
+        assertEq(voters.length, 1);
+        assertEq(voters[0], address(this));
+
+        uint256[] memory activePolls = gogeDao.getActivePolls();
+        assertEq(activePolls.length, 1);
+        assertEq(activePolls[0], 1);
     }
 
+    /// @notice Verify the ability for participants to add votes to a poll via addVote
     function test_gogeDao_addVote_state_change() public {
         test_gogeDao_createPoll();
```

Comment:
Why are we calling tests inside of tests? If there's something here that needs reused this portion should be a utility function not a direct test case.

Reply:
That's correct. This is why I ended up creating create_mock_poll() but never changed older tests. Ultimately it does the same thing